### DISCUSSION TITLE
Hono

### DISCUSSION THREAD

**Discussion by brillout** [vikejs/vike#1391] [Discussion]
Starting a discussion for building an example integrating Hono with Vite and Vike.

Integrating Vike itself is easy since Vike's middleware doesn't depend on Node.js (see [`renderPage()`](https://vike.dev/renderPage)).

So the question boils down to how Hono can be used with Vite.

Prio art: https://github.com/honojs/vite-plugins.

Prior discussion with Hono author and Vite's team: https://github.com/vikejs/vike/discussions/1239#discussioncomment-7482115.

An alternative is to use two different servers: one server for development that uses [`@hono/node-server`](https://github.com/honojs/node-server), and another server for production that doesn't use `@hono/node-server`. Both servers use the same Hono middlewares (both user middlewares and third-party middlewares) and share almost all their code: the only difference is that the dev sever uses `@hono/node-server` and Vite's development middleware, whereas the production server doesn't.

Long-term plan for making Vite's development middelware work everywhere (not only in Node.js):
 - [vitejs/vite#14288](https://github.com/vitejs/vite/discussions/14288)
 - [vitejs/vite#14288 (comment by Hono author)](https://github.com/vitejs/vite/discussions/14288#discussioncomment-7507407)

But it looks like it's going to take a while until that plan is implemented.

Contribution welcome for a Hono + Vite + Vike example.

**Comment by flipvh** [comment]
Thanks for organizing this. I talked with Benton in the Hono discord - general channel about similar stuff.  https://github.com/b3nten/vono
https://discord.com/channels/1011308539819597844/1011308539819597847/1189813864179703808

**Comment by nhattran998** [comment]
Love to hear this <3

**Comment by phonzammi** [comment]
I've tested [vike-hono-example](https://github.com/phonzammi/vike-hono-example). I think it works. Please take a look & review it if you guys are interested. :smile:

**Reply by royalswe** [reply]
Nice done! It runs smooth on macos.

**Reply by nhattran998** [reply]
Thank you <3 Could you try the version to work on the Cloudflare? Would be nice if Vike will work on that.

**Reply by phonzammi** [reply]
@harrytran998, I'm sorry, I'm not familiar with Cloudflare, but I'll try later when I learn more about it
